lib/browser/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { EventEmitter } from 'events'
+
+/**
+* Creates a lazy instance of modules that can't be required before the
+* app 'ready' event by returning a proxy object to mitigate side effects
+* on 'require'
+*
+* @param {Function} creator - Function that creates a new module instance
+* @param {PrototypeHolder} holder - the object holding the module prototype
+* @param {Boolean} isEventEmitter - whether or not the module is an EventEmitter
+* @returns {Object} - a proxy object for the
+*/
+
+type Module<T, B> = B extends true ? EventEmitter & T : T;
+
+export function createLazyInstance <T, B extends boolean> (
+  creator: (() => T),
+  holder: new () => T,
+  isEventEmitter: B
+): Module<T, B> {
+  let lazyModule: T
+  const module: any = {}
+  for (const method in holder.prototype as (keyof T)[]) {
+    module[method] = (...args: any) => {
```

Comment:
This is making the assumption that all keys on the prototype are methods, I believe this is true currently, but to avoid a foot gun here can we either implement by checking `Object.getOwnPropertyDescriptor` and ensuring that all keys on the prototype are indeed methods and throw an error at runtime if they're not.

That way if people try to use this helper in the future with a property they'll have an error to fix instead of wondering where on earth their property is 👍

Reply:
should be set!